src/data/weather.ts

```diff
@@ -179,6 +179,22 @@ const cardinalDirections = [
   "N",
 ];
 
+export function weatherTempRound(
+  value: number | undefined,
+  decimals: number | undefined
+): number | undefined {
+  if (value === undefined) {
+    return undefined;
+  }
+  if (decimals === undefined) {
+    return value;
+  }
+  const rounded = Number.isInteger(value)
+    ? value.toString()
+    : value.toFixed(decimals);
+  return Number(rounded);
```

Comment:
We actually have a util for this. Import from `src/common/number/round.ts`
```suggestion
  return round(value, decimals);
```